Project2015To2017/Reading/ProjectReader.cs

```diff
@@ -32,39 +52,36 @@ public Project Read(string filePath, IProgress<string> progress)
 				return null;
 			}
 
-			XNamespace nsSys = "http://schemas.microsoft.com/developer/msbuild/2003";
-			if (projectXml.Element(nsSys + "Project") == null)
+			var isLegacy = projectXml.Element(XmlLegacyNamespace + "Project") != null;
+			if (!isLegacy)
 			{
 				progress.Report($"This is not a VS2015 project file.");
 				return null;
 			}
 
-			var fileInfo = new FileInfo(filePath);
+			var packageConfig = new NuSpecReader().Read(file, progress);
 
-			var assemblyReferences = LoadAssemblyReferences(projectXml, progress);
-			var projectReferences = LoadProjectReferences(projectXml, progress);
-
-			var packagesConfigFile = FindPackagesConfigFile(fileInfo, progress);
-
-			var packageReferences = LoadPackageReferences(projectXml, packagesConfigFile, progress);
-
-			var includes = LoadFileIncludes(projectXml);
-
-			var packageConfig = new NuSpecReader().Read(fileInfo, progress);
-
-			var projectDefinition = new Project
+			projectDefinition = new Project
 			{
-				FilePath = fileInfo,
-				AssemblyReferences = assemblyReferences,
-				ProjectReferences = projectReferences,
-				PackageReferences = packageReferences,
-				IncludeItems = includes,
+				FilePath = file,
 				PackageConfiguration = packageConfig,
-				PackagesConfigFile = packagesConfigFile,
 				Deletions = Array.Empty<FileSystemInfo>(),
 				AssemblyAttributeProperties = Array.Empty<XElement>()
 			};
 
+			if (EnableCaching)
+			{
+				_cache.Add(hash, projectDefinition);
+			}
+			
+			projectDefinition.AssemblyReferences = LoadAssemblyReferences(projectXml, progress);
```

Comment:
Moved these to be able in future PR to pass `projectDefinition` as parameter instead of `projectXml`.